bids/config.py

```diff
@@ -30,10 +29,14 @@ def set_option(key, value):
     """
     if key not in _settings:
         raise ValueError("Invalid pybids setting: '%s'" % key)
-    # XXX 0.14: Raise error
-    if (key, value) == ("extension_initial_dot", False):
-        warnings.warn("Setting 'extension_initial_dot' to False will be disabled in "
-                      "pybids 0.14", FutureWarning)
+    # XXX 0.16: Remove
+    elif key == "extension_initial_dot":
+        if value is not True:
+            raise ValueError(f"Cannot set {key!r} to {value!r} as of pybids 0.14. "
+                             "This setting is always True, and will be removed "
+                             "entirely in 0.16.")
+        warnings.warn("Setting 'extension_initial_dot' will be removed in pybids 0.16.",
+                      FutureWarning)
```

Comment:
Gave us 2 minor versions to drop this. We could leave it and silently accept `True`, if people prefer...